File: PSC/SERIES/Serie05/Docs/PSC_Y0910T2_LI31N_SE5-1_Attachment_FILES/Cell.java

public abstract class Cell {
  protected abstract char getView();
  public abstract void touch();
  public boolean isBomb() { return false; }

  private enum Stat {
	NONE('*'), FLAG('X'), VIEW(' ');
	public char view;
	Stat(char v) { view=v; }
  }
  Stat stat=Stat.NONE;

  public final void print()        { System.out.print( stat==Stat.VIEW ? getView() : stat.view); } 
  public final boolean isShown()   { return stat==Stat.VIEW; }
  public final boolean isFlagged() { return stat==Stat.FLAG; }
  public final void show()         { stat=Stat.VIEW; }
  public final void toggleFlag()   { 
	  if (stat== Stat.FLAG) stat=Stat.NONE;
	  else
      if (stat== Stat.NONE) stat=Stat.FLAG;
  }
}

File: trunk/PSC/SERIES/Serie05/Src/Cell.c
#include "Cell.h"
enum status {NONE='*', FLAG='X', VIEW=' '};
typedef enum status Status;
typedef struct Tstat{
	Status view;	
}Stat;

Stat stat;

	char getView(){
		puts ("Method not yet implemented."); return 0;
	}
	void touch(){
		puts("Method not yet implemented.");
	}
	boolean isBomb(){return false;}

	void print(){
		printf("%c",(stat.view == VIEW)? getView(): stat.view);
	}
	const boolean isShown(){return stat.view == VIEW;}
	const boolean isFlagged(){return stat.view == FLAG;}
	void show(){stat.view= VIEW;}
	void toggleFlag(){
		if ( stat.view == FLAG ) stat.view=NONE;
		else
		if ( stat.view == NONE ) stat.view=FLAG;
	}
	void Cell_init(){
		stat.view=NONE;
	}
